src/commands/reset.js

```diff
@@ -8,30 +10,76 @@ class ResetCommand extends BaseCommand {
   /**
    * @param {Object} args
    * @param {Object} flags
-   * @param {DockerCompose} dockerCompose
+   * @param {resetSystemConfig} resetSystemConfig
    * @param {Config} config
+   * @param {ConfigCollection} configCollection
+   * @param {DockerCompose} dockerCompose
+   * @param {Docker} docker
+   * @param {tenderdashInitTask} tenderdashInitTask
    * @return {Promise<void>}
    */
   async runWithDependencies(
     args,
     {
       verbose: isVerbose,
+      hard: isHardReset,
+      'platform-only': isPlatformOnlyReset,
     },
-    dockerCompose,
+    resetSystemConfig,
     config,
+    configCollection,
+    dockerCompose,
+    docker,
+    tenderdashInitTask,
   ) {
     const tasks = new Listr([
+      {
+        title: 'Stop services',
+        task: async () => dockerCompose.stop(config.toEnvs()),
+      },
+      {
+        title: 'Reset platform data',
+        enabled: () => isPlatformOnlyReset,
+        task: () => (
+          new Listr([
+            {
+              title: 'Remove platform services and associated data',
+              task: async () => dockerCompose.rmPlatformOnly(config.toEnvs()),
+            },
+            {
+              title: 'Clean up platform volumes',
+              task: async () => docker.pruneVolumes(),
```

Comment:
I guess `down -v` must remove volumes as well

Reply:
`down -v` will remove all containers and named/anonymous volumes created by `up` so that won't work if we are trying to remove `--platform-only` containers. The only way to remove some of the containers is with `rm <name>` but `rm -v` will [only remove anonymous volumes](https://docs.docker.com/compose/reference/rm/), it does not remove named volumes. So these are the options:

- Use `down -v` and do not offer `reset --platform-only` option
- Use `stop` then `rm -v` and change volumes to anonymous
- Use `stop` then `rm` and do `docker volume prune` afterwards to remove named volumes

Reply:
I get it, but I thought you call it in all cases, not just for `--platform-only`

Reply:
I don't understand what you mean. If we call `down -v` in all cases then we need to sync core again, even if we `reset --platform-only`. But the whole point of `--platform-only` is to stop having to wait for core sync when resetting platform. What am I missing here?

Reply:
Not sure that using `pruneVolumes` is a good idea. It will prune all volumes even not related to this docker compose project. What if we just remove volumes using a list of names. I guess it will be safer. Also, if your previous tasks named "Remove platform services and associated data" then I would move this logic there and have just one task

Reply:
Agree that `pruneVolumes` has unintended side effects. But I think collecting a list of volumes and deleting them one by one is an anti-pattern, instead we should be using anonymous volumes for state data and removing with `rm -v`, since all the data we care about (settings and logs) is available through bind mounts.

Reply:
I’m not sure that anonymous volumes are a good idea. It’s easy to lose them but named volumes are a more persistent thing which is matters in the case of node data. Also, I see cases in the future where we are mounting these volumes to different containers.

Why do you think it’s an antipattern. We already have this list in compose config. Do you think the duplication of that list here is a big problem?

Reply:
OK, fair enough. I was only looking at our current use patterns, which always remove the containers and their data simultaneously, which is what anonymous volumes were designed for. Lucky we have you around to plan for the future ;)

I've refactored it to remove the volumes named in the compose file and cleaned up the listr syntax. We still need to use `dockerode` because there is no way to remove named volumes using docker-compose.

Reply:
Hehe... I got your sarcasm :). Now it looks perfect! We even don't need to maintain the whole list of volumes.